src/chai/ManagedArray.inl

```diff
@@ -370,17 +413,25 @@ template<typename T>
 CHAI_INLINE
 CHAI_HOST_DEVICE ManagedArray<T>::operator T*() const {
 #if !defined(__CUDA_ARCH__) && !defined(__HIP_DEVICE_COMPILE__)
-  ExecutionSpace prev_space = m_resource_manager->getExecutionSpace();
-  m_resource_manager->setExecutionSpace(CPU);
-  auto non_const_active_base_pointer = const_cast<T_non_const*>(static_cast<T*>(m_active_base_pointer));
-  m_active_base_pointer = static_cast<T_non_const*>(m_resource_manager->move(non_const_active_base_pointer, m_pointer_record));
-  m_active_pointer = m_active_base_pointer + m_offset;
-
-  m_resource_manager->registerTouch(m_pointer_record);
-
-
-  // Reset to whatever space we rode in on
-  m_resource_manager->setExecutionSpace(prev_space);
+  if (m_active_pointer) {
+     if (m_resource_manager == nullptr) {
+        m_resource_manager = ArrayManager::getInstance();
+     }
+     if (m_pointer_record == nullptr || m_pointer_record == &ArrayManager::s_null_record) {
+        CHAI_LOG(Warning, "nullptr pointer_record associated with non-nullptr active_pointer")
+     }
+     ExecutionSpace prev_space = m_resource_manager->getExecutionSpace();
+     m_resource_manager->setExecutionSpace(CPU);
+     auto non_const_active_base_pointer = const_cast<T_non_const*>(static_cast<T*>(m_active_base_pointer));
+     m_active_base_pointer = static_cast<T_non_const*>(m_resource_manager->move(non_const_active_base_pointer, m_pointer_record));
```

Comment:
If we have nested critters, do they get moved here?

Reply:
calling move instead now.